cmd/provisioner-localpv/app/config.go

```diff
@@ -170,15 +190,21 @@ func (p *Provisioner) GetVolumeConfig(ctx context.Context, pvName string, pvc *c
 	}
 
 	//TODO : extract and merge the cas volume config from pvc
-	//This block can be added once validation checks are added
+	// This block can be added once validation checks are added
 	// as to the type of config that can be passed via PVC
-	//pvcCASConfigStr := pvc.ObjectMeta.Annotations[string(mconfig.CASConfigKey)]
-	//if len(strings.TrimSpace(pvcCASConfigStr)) != 0 {
-	//	pvcCASConfig, err := cast.UnMarshallToConfig(pvcCASConfigStr)
-	//	if err == nil {
-	//		pvConfig = cast.MergeConfig(pvcCASConfig, pvConfig)
-	//	}
-	//}
+	pvcCASConfigStr := pvc.ObjectMeta.Annotations[string(mconfig.CASConfigKey)]
+	klog.V(4).Infof("PVC %v has config:%v", pvc.Name, pvcCASConfigStr)
+	if len(strings.TrimSpace(pvcCASConfigStr)) != 0 {
+		pvcCASConfig, err := cast.UnMarshallToConfig(pvcCASConfigStr)
+		if err == nil {
+			pvConfig = cast.MergeConfig(pvcCASConfig, pvConfig)
```

Comment:
Here pvcCASConfig is essentially the scCASConfig from the previous block. The SC config (cluster admin's configuration) should have precedence over the PVC config (cluster user's configuration). User shouldn't be able to override admin's config.

```suggestion
			pvConfig = cast.MergeConfig(pvConfig, pvcCASConfig)
```